pkg/sql/alter_function.go

```diff
@@ -210,32 +210,20 @@ func (n *alterFunctionRenameNode) startExec(params runParams) error {
 		}
 	}
 
-	// Disallow renaming if this rename operation will break other UDF's invoking
-	// this one.
-	var dependentFuncs []string
-	for _, dep := range fnDesc.GetDependedOnBy() {
-		desc, err := params.p.Descriptors().ByIDWithoutLeased(params.p.Txn()).Get().Desc(params.ctx, dep.ID)
-		if err != nil {
-			return err
-		}
-		_, ok := desc.(catalog.FunctionDescriptor)
-		if !ok {
-			continue
-		}
-		fullyResolvedName, err := params.p.GetQualifiedFunctionNameByID(params.ctx, int64(dep.ID))
-		if err != nil {
-			return err
-		}
-		dependentFuncs = append(dependentFuncs, fullyResolvedName.FQString())
+	// Disallow renaming if this rename operation will break other UDF's or views
+	// invoking this one.
+	dependentObjects, err := getFuncRefsDisallowingAlter(params, fnDesc)
+	if err != nil {
+		return err
 	}
-	if len(dependentFuncs) > 0 {
+	if len(dependentObjects) > 0 {
 		return errors.UnimplementedErrorf(
 			errors.IssueLink{
 				IssueURL: build.MakeIssueURL(83233),
 				Detail:   "renames are disallowed because references are by name",
 			},
-			"cannot rename function %q because other functions ([%v]) still depend on it",
-			fnDesc.Name, strings.Join(dependentFuncs, ", "))
+			"cannot rename function %q because other functions or views ([%v]) still depend on it",
```

Comment:
if we complete [`7edd9f0` (#146475)](https://github.com/cockroachdb/cockroach/pull/146475/commits/7edd9f05bdc8f5a874d1bcf3e234ed2fc6a5e6b5) then there would be no need to block the function from being renamed even if a view references it.

i don't mind tackling that separately after this PR gets merged, but a comment mentioning this followup work might be useful.